cowprotocol/cowswap
# Custom GA dimension implementation

# Summary

This PR fixes the issue of custom dimensions with a hack solution because the `set` method doesn't work as expected.
With this method we store the custom dimensions in a GoogleAnalyticsProvider class and then pass those dimensions as event parameters.
The reason we pass parameter names such as `dimension1` instead of `chainId` is because that is what plugin expects

# To test
1. Make sure you have GA debugger on
2. Connect to some wallet or change the network
3. Easy event to test is to just switch INPUT/OUTPUT tokens on the arrow button in the swap component
4. This will create `Switch INPUT/OUTPUT Tokens` event and make sure there are dimension parameter passed with correct values
5. `chainId` is `dimension1` and `walletName` is `dimension2` and `customBrowserType` is `dimension3`

![Screenshot from 2022-09-28 11-25-10](https://user-images.githubusercontent.com/34926005/192745054-7ae23a18-f868-4df6-bf08-5ccfcc3053a3.png)

**From the GA console, seems it worked**
![Screenshot from 2022-09-29 11-09-52](https://user-images.githubusercontent.com/34926005/192993525-cd9eaf68-6770-4c62-b6c5-7d2cbae6a5e5.png)
